wheel/wheelfile.py

```diff
@@ -63,12 +77,12 @@ def __init__(self, file, mode='r'):
                     path, hash_sum, size = line.rsplit(u',', 2)
                     if hash_sum:
                         algorithm, hash_sum = hash_sum.split(u'=')
-                        if algorithm not in hashlib.algorithms_available:
```

Comment:
If you have a working function to check availability, why did you still move the check elsewhere?

Reply:
It's still done in the constructor, it's just that IMHO, in the unlikely event that `md5`,/ `sha` is unsupported, the code should still complain about PEP 427, not about the missing support.